src/glance.gleam

```diff
@@ -273,8 +273,14 @@ pub type Variant {
   Variant(name: String, fields: List(Field(Type)))
 }
 
+pub type RecordUpdateField(t) {
+  RecordUpdateField(label: String, item: Option(t))
+}
+
 pub type Field(t) {
-  Field(label: Option(String), item: t)
+  LabelledField(label: String, item: t)
+  ShorthandField(label: String)
```

Comment:
This can't be used for `CustomType` definitions now, they cannot use the shorthand syntax

Reply:
I'll add a test to make sure that isn't allowed. I'm not sure of a nice way to structure `Field` itself to account for that though

Reply:
Oh, apparently I already did that with `no_shorthand_in_constructors_test()`

Reply:
Yeah, so far my solution is by using a separate `explicit_field` function in `optional_variant_fields`:

```gleam
fn optional_variant_fields(
  tokens: Tokens,
) -> Result(#(List(Field(Type)), Tokens), Error) {
  case tokens {
    [#(t.LeftParen, _), #(t.RightParen, _), ..tokens] -> Ok(#([], tokens))
    [#(t.LeftParen, _), ..tokens] -> {
      comma_delimited(
        [],
        tokens,
        // ensure a label shorthand field is not used in a record constructor definition
        explicit_field(_, of: type_),
        until: t.RightParen,
      )
    }
    _ -> Ok(#([], tokens))
  }
}
```

```gleam
fn explicit_field(
  tokens: Tokens,
  of parser: fn(Tokens) -> Result(#(t, Tokens), Error),
) {
  use #(t, tokens) <- result.try(field(tokens, parser))

  case t {
    ShorthandField(..) -> unexpected_error(tokens)
    _ -> Ok(#(t, tokens))
  }
}
```

Not really the most type-safe approach, but I'm not sure how else to tackle this without introducing a third field type.

Reply:
We could just add a `VariantField` type the same way we added a `RecordUpdateField`, it just feels a bit disorganized having multiple types that perform nearly the same task.

Reply:
They are similar, but they are different.

The problem is not that it might be parsed that way, it's the types no longer accurately reflect the restrictions of the data.

Reply:
I got distracted for a while there, but I've now added a new `VariantField` type.

Hopefully it's what you're looking for!